public/app/features/datasources/state/actions.ts

```diff
@@ -264,13 +265,18 @@ export function loadDataSourcePlugins(): ThunkResult<void> {
   };
 }
 
+const dsApiVersions = new DatasourceAPIVersions();
+
 export function updateDataSource(dataSource: DataSourceSettings) {
   return async (
     dispatch: (
       dataSourceSettings: ThunkResult<Promise<DataSourceSettings>> | { payload: unknown; type: string }
     ) => DataSourceSettings
   ) => {
     try {
+      if (config.featureToggles.grafanaAPIServerWithExperimentalAPIs) {
+        dataSource.apiVersion = await dsApiVersions.get(dataSource.type);
```

Comment:
So this means the backend dictates what the preferred version is? How would the datasource config UI be able to adapt to different versions if this happens on update? I might be completely out of the loop so forgive me if I've misunderstood something.

Reply:
> So this means the backend dictates what the preferred version is?

Correct.

> How would the datasource config UI be able to adapt to different versions if this happens on update?

I don't think we plan to do so, it's like the UI serving two different major versions of the same plugin.